kafka/src/test/java/org/apache/apex/malhar/kafka/KafkaInputOperatorTest.java

```diff
@@ -232,33 +228,11 @@ public void endWindow()
       int tupleSize = windowTupleCollector.size();
       tupleCollection.addAll(windowTupleCollector);
       
-      int countDownTupleSize = countDownAll ? tupleSize : endTuples;
+      int countDownTupleSize = endTuples;
 
       if (latch != null) {
-        Assert.assertTrue("received END_TUPLES more than expected.", latch.getCount() >= countDownTupleSize);
-        while (countDownTupleSize > 0) {
+        while (countDownTupleSize-- > 0) {
             latch.countDown();
-            --countDownTupleSize;
-        }
-        if (latch.getCount() == 0) {
-          /**
-           * The time before countDown() and the shutdown() of the application
```

Comment:
I don't understand this part. It says it will cause Catastrophic Error if you change alive value but alive can only be set to false by shutdown method which shouldn't cause this error.